preinstall.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+echo "Setting up the development environment..."
+
+# Function to install a package if not already installed
+install_package_linux() {
+  PACKAGE=$1
+  # Check for apt-get availability
+  if ! command -v apt-get &> /dev/null; then
+    echo "apt-get not found. Please install packages manually."
+    exit 1
+  fi
+  if ! command -v $PACKAGE &> /dev/null; then
+    echo "Installing $PACKAGE on Linux..."
+    sudo apt-get update -y
```

Comment:
Thoughts: So there is an issue with this. It assumes that the Linux distro being used is either Ubuntu or Debian based. There a quite a few different variants of Linux that don't use `apt` as the package manager. The other popular variant is Fedora/Redhat which uses the `dnf` command. We may want to add this but I would say most people on Linux would be able to install this manually anyway.

Reply:
Thanks for pointing it out. Updated!

at this moment for Linux distros only apt and dnf package managers are supported. For mac only brew is handled. This should cover most of developers. Otherwise they need to install them manually.

Reply:
Looks good!